cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -361,29 +315,6 @@ func getControllerManagerCommand(cfg *kubeadmapi.ClusterConfiguration) []string
 		defaultArguments["cluster-name"] = cfg.ClusterName
 	}
 
-	// TODO: The following code should be remvoved after dual-stack is GA.
-	// Note: The user still retains the ability to explicitly set feature-gates and that value will overwrite this base value.
-	enabled, present := cfg.FeatureGates[features.IPv6DualStack]
-	if present {
-		defaultArguments["feature-gates"] = fmt.Sprintf("%s=%t", features.IPv6DualStack, enabled)
-	}
-	if cfg.Networking.PodSubnet != "" {
```

Comment:
since we validate that the node-masks are correct , we must not change the masks later

Reply:
i guess this is true, but we need to double check that we are backwards compatible.

the docs for the flag `--node-cidr-mask-size` say:
> Mask size for node cidr in cluster. Default is 24 for IPv4 and 64 for IPv6.

https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/

so i guess if the user is not passing the flag explicitly it should be good for them and with kubeadm's default podSubNet (that has `/16`)

Reply:
Yeah, for validation we use the user configured, or the default depending on the IP family and the dual stack flag.
We don't need to pass the default to the controller manager because is redundant

Reply:
though this seems like a breaking change to users that are using IPV6 and don't know how to calculate the IPV6 masks. 
the default value is 64 there and they will get validation error now.

```
case podNetSize == 112:
  // Special case, allows 256 nodes, 256 pods/node
  nodeCidrSize = 120
case podNetSize < 112:
  // Use multiple of 8 for node CIDR, with 512 to 64K nodes
 nodeCidrSize = totalBits - ((totalBits-podNetSize-1)/8-1)*8
```

so previously if they were passing `2001:db8:1::/112` for podsubnet they would get `--node-cidr-mask-size=120` and if they were passing `some-ipv6-ip/111`, they would get `--node-cidr-mask-size=128`.

given ipv6 support in k8s is shaky due to the bitmap problems we might go with this and add `action required:` in the release note...

Reply:
in the other hand, you see what happen with kind, we were changing a configuration option without telling the user.
The user has to known in advance how to subnet their cluster, the CNI will require this info.

Reply:
The previous experience of kubeadm making the change for the user had it's problems. It generally lead to confusion, so I'm +1 for making this change. 
However, it will be breaking change for some users. I'm okay with a release note documenting this.

Reply:
> 

@Arvinderpal How does user recover from this breaking change then?

Reply:
what do you mean exactly @aoxn ?

the change was done in an alpha feature in kubernetes 1.20 so the change is justified, are you upgrading from 1.19?

 you need to set the `--node-cidr-mask-size=` accordingly if the default value doesn't work for you, which part is failing to you?